plugins/base/src/main/kotlin/renderers/html/SearchbarDataInstaller.kt

```diff
@@ -4,74 +4,107 @@ import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
 import org.jetbrains.dokka.Platform
 import org.jetbrains.dokka.model.DisplaySourceSet
 import org.jetbrains.dokka.model.dfs
+import org.jetbrains.dokka.model.withDescendants
 import org.jetbrains.dokka.pages.*
+import org.jetbrains.dokka.plugability.DokkaContext
+import org.jetbrains.dokka.transformers.pages.PageTransformer
 import java.util.concurrent.ConcurrentHashMap
 
-data class SearchRecord(val name: String, val description: String? = null, val location: String, val searchKeys: List<String> = listOf(name)) {
-    companion object { }
+typealias PageId = String
+typealias Json = String
+
+data class SearchRecord(
+    val name: String,
+    val description: String? = null,
+    val location: String,
+    val searchKeys: List<String> = listOf(name)
+) {
+    companion object {}
 }
 
-open class SearchbarDataInstaller {
-    private val mapper = jacksonObjectMapper()
+open class SearchbarDataInstaller(val context: DokkaContext) : PageTransformer {
+    data class PageWithId(val id: PageId, val page: ContentPage) {
+        val displayableSignature = getSymbolSignature(page)?.let { flattenToText(it) } ?: page.name
+    }
 
-    private val pageList = ConcurrentHashMap<String, Pair<String, String>>()
+    private val mapper = jacksonObjectMapper()
 
-    open fun generatePagesList(): String {
-        val pages = pageList.entries
+    open fun generatePagesList(pages: Map<PageId, PageWithId>, locationResolver: PageResolver): Json =
```

Comment:
I think returning there Json implies that you need to use jackson. String was more flexible

Reply:
I don't think it does since JSON is just an alias for String